js/src/Boxplot.ts

```diff
@@ -26,17 +26,23 @@ export class Boxplot extends Mark {
 
     render() {
         const base_creation_promise = super.render.apply(this);
-        const that = this;
 
         this.selected_style = this.model.get("selected_style");
         this.unselected_style = this.model.get("unselected_style");
 
-        return base_creation_promise.then(function() {
-            that.event_listeners = {};
-            that.create_listeners();
-            that.process_interactions();
-            that.draw();
-        }, null);
+        this.display_el_classes = ["box"];
+
+        this.displayed.then(() => {
+            this.parent.tooltip_div.node().appendChild(this.tooltip_div.node());
+            this.create_tooltip();
+        });
+
+        return base_creation_promise.then(() => {
+            this.event_listeners = {};
+            this.create_listeners();
+            this.process_interactions();
+            this.draw();
+        });
```

Comment:
Maybe use await?, and make render async?

Reply:
👍🏼